app-backend/backsplash-core/src/main/scala/com/rasterfoundry/backsplash/BacksplashImage.scala

```diff
@@ -387,91 +363,88 @@ sealed abstract class MultiTiffImage[F[_]: Monad, G[_]](
       }
     }
 
-  def readWithCache(z: Int, x: Int, y: Int, context: TracingContext[F])(
-      implicit @cacheKeyExclude flags: Flags
-  ): F[Option[MultibandTile]] = {
+  def readWithTrace(z: Int,
+                    x: Int,
+                    y: Int,
+                    context: TracingContext[F]): F[Option[MultibandTile]] = {
     val readTags = tags.combine(Map("zoom" -> z.toString))
-    context.childSpan("cache.read:z_x_y:", readTags) use { child =>
-      memoizeF(None) {
-        val layoutDefinition = BacksplashImage.tmsLevels(z)
-        for {
-          sources <- subsetBands.toNel match {
-            case Some(bs) => getBandRasterSources(bs, child)
-            case None =>
-              Err.raiseError(
-                RequirementFailedException("Must request at least one band")
-              )
-          }
-          tiles <- sources parTraverse { rs =>
-            child.childSpan("readFromRasterSource", tagRasterSource(rs)) use {
-              _ =>
-                sync.delay {
-                  rs.reproject(WebMercator)
-                    .tileToLayout(layoutDefinition, NearestNeighbor)
-                    .read(SpatialKey(x, y), List(0)) map { tile =>
-                    tile
-                      .mapBands((_: Int, t: Tile) => t.toArrayTile)
-                      .band(0)
-                  }
+    context.childSpan("read:z_x_y:", readTags) use { child =>
+      val layoutDefinition = BacksplashImage.tmsLevels(z)
+      for {
+        sources <- subsetBands.toNel match {
+          case Some(bs) => getBandRasterSources(bs, child)
+          case None =>
+            Err.raiseError(
+              RequirementFailedException("Must request at least one band")
+            )
+        }
+        tiles <- sources parTraverse { rs =>
+          child.childSpan("readFromRasterSource", tagRasterSource(rs)) use {
+            _ =>
+              sync.delay {
+                rs.reproject(WebMercator)
+                  .tileToLayout(layoutDefinition, NearestNeighbor)
+                  .read(SpatialKey(x, y), List(0)) map { tile =>
+                  tile
+                    .mapBands((_: Int, t: Tile) => t.toArrayTile)
+                    .band(0)
                 }
-            }
+              }
           }
-          mbt <- child.childSpan("constructMultibandTile", readTags) use { _ =>
-            app.pure {
-              Some(MultibandTile(tiles.toList collect { case Some(t) => t }))
-            }
+        }
+        mbt <- child.childSpan("constructMultibandTile", readTags) use { _ =>
+          app.pure {
+            Some(MultibandTile(tiles.toList collect { case Some(t) => t }))
           }
-        } yield mbt
-      }
+        }
+      } yield mbt
     }
   }
 
```

Comment:
The `readWithCache` functions existed so we could use flags to trigger the caches on/off with flags set via environment variables. I would delete these functions and put the logic in `read` now -- otherwise these extra functions add a level of indirection.